.github/workflows/go-test.yml

```diff
@@ -19,7 +19,6 @@ jobs:
           - ubuntu-latest
           - macos-latest
         go-version:
-          - 1.22.x
```

Comment:
It looks like that this version could be removed since minimum go version now is 1.23.0

Looking at the github action output for 1.22, at the end it uses 1.23.6 too:
https://github.com/elastic/go-resource/actions/runs/13288641101/job/37103506890?pr=27#step:5:18

Reply:
Comparing with a previous action, previously it was testing with go 1.22.11
https://github.com/elastic/go-resource/actions/runs/13261332035/job/37018427230#step:5:17